src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -245,15 +245,17 @@ internal static IEnumerable<Claim> MergeClaims(IEnumerable<Claim> claims, IEnume
 
         /// <summary>
         /// Check whether the given exception type is recoverable by LKG.
+        /// Decryption error is only recoverable, if the configuration has decryption keys in it.
         /// </summary>
         /// <param name="exception">The exception to check.</param>
+        /// <param name="configContainsDecryptionKeys">Whether the configuration contains decryption keys.</param>
         /// <returns><c>true</c> if the exception is certain types of exceptions otherwise, <c>false</c>.</returns>
-        internal static bool IsRecoverableException(Exception exception)
+        internal static bool IsRecoverableException(Exception exception, bool configContainsDecryptionKeys)
         {
             return exception is SecurityTokenInvalidSignatureException
                   || exception is SecurityTokenInvalidIssuerException
                   || exception is SecurityTokenSignatureKeyNotFoundException
-                  || exception is SecurityTokenDecryptionFailedException;
+                  || (exception is SecurityTokenEncryptionKeyNotFoundException && configContainsDecryptionKeys);
```

Comment:
SecurityTokenEncryptionKeyNotFoundException seems to be to narrow. SecurityTokenDecryptionFailedException looks like a better fit for the scenario.

Reply:
Would you want to trigger the LKG in other conditions that KeyNotFound, @GeoK ? what would the scenario be?

Reply:
Updated the code:
- Don't try to match on key IDs
- Revert to throwing a more general token decryption error

We still only do the refresh if the current config has keys.